Update README (#123)
signalfx/splunk-otel-java: README.md

> :warning: Specify the agent as the only JVM agent for your application.
> Multiple agents may result in unpredictable results, broken instrumentation,
> and in some cases might crash your application.

To see the Java Agent in action with sample applications, see our
[examples](https://github.com/signalfx/tracing-examples/tree/master/opentelemetry-tracing/opentelemetry-java-tracing).

### Basic Configuration

The service name resource attribute is the only configuration option
that typically needs to be specified. You can set it by adding a `service.name`
attribute as shown in the [example above](#getting-started).

A couple other configuration options that may need to be changed or set are:

- Endpoint if not sending to a locally running Smart Agent with default
  configuration. See the [Jaeger exporter](#jaeger-exporter) section for more information.
- Environment resource attribute (example:
  `-Dotel.resource.attributes=service.name=my-service,environment=production`) to specify what
  environment the span originated from.

### Supported Java Versions

The agent works with Java runtimes version 8 and higher and supports all

> JVM-based languages (for example, Clojure, Groovy, Kotlin, Scala). Supported

Reviewer: Out of curiousity, has anyone tried using otel with Clojure? I'd love to see an example of that in action.

| ------------------------------------ | ----------------------------------   | -------------- | -------------------------------------------------------- |
| splunk.context.server-timing.enabled | SPLUNK_CONTEXT_SERVER_TIMING_ENABLED | false          | Enables adding `Server-Timing` header to HTTP responses. |

### More options
For more options see [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation#configuration-parameters-subject-to-change)

> ## Span tags the Splunk distribution automatically creates

Reviewer 2: I vaguely remember somebody wanted to see all the attributes that our distribution adds on top of the standard Otel...

Reviewer 3: Maybe we can move them to a separate doc and reference it from the main README?

To extend the instrumentation with the OpenTelemetry Instrumentation for Java,
you have to use a compatible API version. The Splunk distribution of
OpenTelemetry Java Instrumentation version 0.8.0 is compatible with the
OpenTelemetry Instrumentation for Java version 0.16.1 and API version 0.16.0.

## Correlating traces with logs

To correlate traces with logs it is possible to add both trace (`traceId` and
`spanId`) and resource (e.g., `service.name` and `environment` attributes of
the
[Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/sdk.md))

> contexts.

Reviewer: this sentence is pretty mangled. I'm not sure what it's trying to communicate.

> To correlate traces with logs it is possible to add both trace (`traceId` and
> `spanId`) and resource (e.g., `service.name` and `environment` attributes of
> the
> [Resource](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/sdk.md))
> contexts.

Author: ```suggestion
To correlate traces with logs it is possible to add the following metadata from traces to logs:

 - Trace: `traceId` and `spanId`
 - Resource: `service.name` and `environment`
```

Documentation on how to inject trace context into logs is available
[here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/docs/logger-mdc-instrumentation.md).

> To log resource context, Splunk distribution exposes resource attributes as

Reviewer: ```suggestion
To log resource context, the Splunk distribution exposes resource attributes as
```
